crates/pathfinder/src/rpc/v02/method/get_transaction_by_block_id_and_index.rs

```diff
@@ -0,0 +1,256 @@
+use anyhow::Context;
+
+use crate::core::{BlockId, StarknetTransactionIndex};
+use crate::rpc::v02::types::reply::Transaction;
+use crate::rpc::v02::RpcContext;
+use crate::storage::{StarknetBlocksBlockId, StarknetBlocksTable, StarknetTransactionsTable};
+
+#[derive(serde::Deserialize, Debug, PartialEq, Eq)]
+pub struct GetTransactionByBlockIdAndIndexInput {
+    block_id: BlockId,
+    index: StarknetTransactionIndex,
+}
+
+crate::rpc::error::generate_rpc_error_subset!(
+    GetTransactionByBlockIdAndIndexError: BlockNotFound,
+    InvalidTxnIndex
+);
+
+#[allow(dead_code)]
+pub async fn get_transaction_by_block_id_and_index(
+    context: std::sync::Arc<RpcContext>,
+    input: GetTransactionByBlockIdAndIndexInput,
+) -> Result<Transaction, GetTransactionByBlockIdAndIndexError> {
+    let index: usize = input
+        .index
+        .get()
+        .try_into()
+        .map_err(|_| GetTransactionByBlockIdAndIndexError::InvalidTxnIndex)?;
+
+    let block_id = match input.block_id {
+        BlockId::Hash(hash) => hash.into(),
+        BlockId::Number(number) => number.into(),
+        BlockId::Latest => StarknetBlocksBlockId::Latest,
+        BlockId::Pending => {
+            return get_transaction_from_pending(&context.pending_data, index).await
+        }
+    };
+
+    let storage = context.storage.clone();
+    let span = tracing::Span::current();
+
+    let jh = tokio::task::spawn_blocking(move || {
```

Comment:
Not related to this PR, just got me thinking.

Wonder if we shouldn't make this a function of sorts to not forget the span.. maybe not worth it though? Could also fold in the database transaction creation? Although maybe that's just weird.